app/models/folder_searcher.rb

```diff
@@ -0,0 +1,37 @@
+class FolderSearcher
+
+  def initialize(user, params: {})
+    @user = user
+    @params = params
+  end
+
+  def search
+    query = 'attachment_file_name ILIKE :search OR description ILIKE :search'
+    search = "%#{search_term.strip}%"
+    attachment = Attachment.where(query, search: search ).map do |a|
```

Comment:
Is this iterating over the result and removing locked resources? Perhaps `select` is more direct than `map` and `compact`?